src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -24,21 +24,37 @@ protected RemoteHostClient(Workspace workspace)
 
         public event EventHandler<bool> ConnectionChanged;
 
+        /// <summary>
+        /// Create <see cref="RemoteHostClient.Session"/> for the <paramref name="serviceName"/> if possible.
+        /// otherwise, return null.
+        /// </summary>
```

Comment:
Future consumers would likely find it helpful if doc comment gave examples of when this can be null (make it clear whether it's due to configuration vs error).